plugin/math-katex/math-katex.js

```diff
@@ -0,0 +1,523 @@
+			'.reveal section[data-math]'
+		);
+
+
+		/**
+		 * Tests if an element should be ignored for formula replacements.
+		 */
+		function isIgnored( element ) {
+
+			var e = element;
+
+			var isIgnoredElement =
+				options.ignoredElements.indexOf( e.nodeName ) !== -1;
+
+			// Elements may be marked as ignored with a class
+			var isIgnoredClass =
+				e.classList.contains( defaults.ignoredClass );
+
+			// Also look for the ignored class on the parent (non-recursive)
+			var parent = e.parentNode;
+			if ( parent ) {
+				isIgnoredClass = isIgnoredClass || parent.classList.contains(
+					defaults.ignoredClass
+				);
+			}
+
+			// Ignore script elements, unless they are templates (e.g. Markdown)
+			var isTemplate = e.getAttribute( 'type' ) === 'text/template';
+			var isScript = e.nodeName === 'SCRIPT' && !isTemplate;
+
+			return isIgnoredElement || isIgnoredClass || isScript;
+		}
+
+
+		/**
+		 * `forEach` for a set of DOM elements, excludes ignored elements.
+		 */
+		function each( arrayLike, f ) {
+			for ( var i = 0; i < arrayLike.length; i++ ) {
+				var element = arrayLike[i];
+
+				if ( !isIgnored( element ) ) {
+					f( element, i );
+				}
+			}
+		}
+
+
+		// Render <… class="formula"> … <…/> formulas
+
+		each( wrappedElements, function ( e ) {
+
+			var formula = e.innerHTML;
+			var offset = 0;    // For error-position correction
+
+			if (e.classList.contains( 'display' )) {
+				// Prepend KaTeX instruction, correct offset
+
+				var prefix = '\\displaystyle {';
+				formula = prefix + formula + '}';
+				offset -= prefix.length;
+			}
+			else {
+				e.classList.add( 'inline' );     // ensure class
+			}
+
+			try {
+				e.innerHTML = katex.renderToString( preProcess( formula ) );
+
+				e.classList.add( 'formula' );    // In case it was selected
+				                                 // with `.math`.
+
+				// KaTeX adds a wrapper element, but we already have one, so
+				// unwrap:
+				e.classList.add( 'katex' );
+				e.innerHTML = e.querySelector( '.katex' ).innerHTML;
+			}
+			catch ( error ) {
+				correctErrorPosition( error, offset );  // mutates the error
+				handleError( error, e );
+			}
+		});
+
+
+		// Render `$…$` and `$$…$$` formulas
+		// (must run after wrapped elements replacements to not replace twice)
+
+		each( texSlideElements, function ( e ) {
+
+			try {
+				e.innerHTML = replaceFormulaTex( 'display', e.innerHTML );
+				e.innerHTML = replaceFormulaTex( 'inline',  e.innerHTML );
+				e.innerHTML = unescapeDollarSign( e.innerHTML );
+
+				// Add a class to the created rendered elements (but keep class
+				// `katex`, KaTeX needs this for its own CSS)
+
+				var renderedFormulas = e.querySelectorAll( '.katex' );
+
+				each( renderedFormulas, function ( forumula ) {
+					forumula.classList.add( defaults.formulaClass );
+				});
+			}
+			catch ( error ) {
+				handleError( error, e );
+			}
+		});
+	}
+
+
+	// --- Script and stylesheet loading ---------------------------------------
```

Comment:
Yeah, I’d like to get rid of that function as well, but if I’m not missing something then `head.load` would give me no error callbacks.